types/types.go

```diff
@@ -633,6 +633,8 @@ type SystemContext struct {
 	// === dir.Transport overrides ===
 	// DirForceCompress compresses the image layers if set to true
 	DirForceCompress bool
+	// DirForceDecompress decompresses the image layers if set to true
+	DirForceDecompress bool
```

Comment:
Cc: @vrothberg
- Should this be just a `dir:` feature or a generic `c/image/copy.Options` option? I don’t have much of an idea whether this is likely to be used elsewhere. It doesn’t _hurt_ here, doing it requires much less thinking about designing the `copy.Options` value _well_ (interaction of that new option with `DesiredLayerCompression` would be a concern), and this is cheap to implement/maintain even if we later decided to make this a `copy` feature, so I’m leaning towards this `dir:`-specific approach — but if there were already a good reason to make this a `copy` feature _now_, please speak up.
- RFC: I think I’d prefer a `DirDesiredLayerCompression *types.LayerCompression`, with the default `nil` meaning “unspecified”, and document that `DirForceCompress` is ignored (forbidden?) if `DirDesiredLayerCompression` is specified. That would naturally extend to future values of `LayerCompression`, if any, and it avoids the question of “what does `DirForceCompress ==true && DirForceDecompress == true` mean?”. I don’t feel too strongly about this, we can just reject the true && true combination instead.

Reply:
> but if there were already a good reason to make this a copy feature now, please speak up.

I can only think of one reasons: determinism.  Copying decompressed layers around would always yield the same result while compressing layers may yield different results depending on the (version of the) tools and the tool chains.

Maybe via `--dest-compress-level=0`?

Reply:
`--dest-compress-level` does not have globally defined semantics, and there are no reserved values; the Gzip compressor already recognizes the `0` value (as “store in the Gzip envelope with no compression applied” it seems).

Reply:
Deterministically storing ~twice as much data on a registry… I don’t know. A hypothetical auditor can check RootFS.DiffIDs in most cases.

Reply:
Do I understand correctly that this PR could in principle be merged with the current concept, the assumption being that a follow-up PR might always happen at a later point to extend the functionality to other protocols?

If so I can start working at the other comments here.

Reply:
Yeah, let’s go with the `dir:`-only approach for now.